src/db-adapters/Mongoose/MongooseAdapter.js

```diff
@@ -614,4 +591,27 @@ export default class MongooseAdapter {
 
     return words.join(" ");
   }
+
+  static makeIdQuery(idOrIds) {
+    const idIsValid = mongoose.Types.ObjectId.isValid;
+    let idQuery;
+
+    if (typeof idOrIds === "string") {
+      if (!idIsValid(idOrIds)) {
+        throw new APIError(404, undefined, "No matching resource found.");
+      }
+
+      idQuery = idOrIds;
+    } else if (Array.isArray(idOrIds)) {
+      let validIds = idOrIds.filter(idIsValid);
```

Comment:
Silently filtering out invalid ids seems like a bad idea. Instead, if any of the ids are invalid, I think we should 404.